src/core_modules/capture-core/components/Pages/MainPage/WorkingListsCommon/hooks/useColumns.js

```diff
@@ -1,17 +1,17 @@
 // @flow
 import { useMemo } from 'react';
 import type { CustomColumnOrder } from '../../WorkingListsCommon';
-import type { EventWorkingListsColumnConfigs } from '../types';
 
-export const useColumns = (
-    customColumnOrder?: CustomColumnOrder,
-    defaultColumns: EventWorkingListsColumnConfigs,
-): EventWorkingListsColumnConfigs => {
+export const useColumns = <TColumnConfigs: Array<{ id: string, visible: boolean, [string]: any }>>(
+    customColumnOrder?: CustomColumnOrder, // eslint-disable-line
+    defaultColumns: TColumnConfigs,
+): TColumnConfigs => {
     const defaultColumnsAsObject = useMemo(() =>
         defaultColumns
             .reduce((acc, column) => ({ ...acc, [column.id]: column }), {}),
     [defaultColumns]);
 
+    // $FlowFixMe Based on the business logic the customColumOrder id should exists as a key in defaultColumnsAsObject
```

Comment:
cant say I  understand 100 percent the use case here, but by doing this you could remove the flow fix me 

```
type Config = Array<{ id: string, visible: boolean, [string]: any }>

export const useColumns = <TColumnConfigs: Config>(
    customColumnOrder?: CustomColumnOrder, // eslint-disable-line
    defaultColumns: Config,
): Config => {
    const defaultColumnsAsObject = useMemo(() =>
        defaultColumns
            .reduce((acc, column) => ({ ...acc, [column.id]: column }), {}),
    [defaultColumns]);

    return useMemo(() => {
        if (!customColumnOrder) {
            return defaultColumns;
        }
.
.
.
 
```

Reply:
The point here is that I want to return the type that is passed in as `defaultColumns` (the more specific one, not the more generic `Config`). For example from the `EventWorkingListsColumnSetup` file:

![image](https://user-images.githubusercontent.com/12745737/101164883-02eaac80-3636-11eb-9154-4d3d4c2e83b5.png)

You will see that flow recognizes the return value as `EventWorkingListsColumnConfigs`, if we do what you suggest here the return value will be the more generic type `Config`.

Reply:
Ow now I get it! Thanks!!